Drivers/Src/Lights.c

```diff
@@ -9,16 +9,60 @@
 */
 
 
-
 #include "Lights.h"
+#include "BSP_SPI.h"
+#include "stm32f4xx.h"
+#include "Tasks.h"
+
+//static State lightStates[10] = {OFF};
+static uint16_t lightStatesBitmap;
+static uint16_t lightToggleBitmap;
+static OS_MUTEX lightMutex;
+static CPU_TS timestamp;
+static OS_ERR err;
 
 /**
 * @brief   Initialize Lights Module
 * @return  void
 */ 
 void Lights_Init(void) {
+    OSMutexCreate(&lightMutex, "Light lock", &err); //init mutex
+    assertOSError(OS_BLINK_LIGHTS_LOC, err);
+
+    OSMutexPend(&lightMutex,0,OS_OPT_PEND_BLOCKING,&timestamp,&err); //lock light hardware for initialization (TODO: may not be needed, just here for additional safety)
+    assertOSError(OS_BLINK_LIGHTS_LOC, err);
+
     BSP_GPIO_Init(LIGHTS_PORT, 0x3C0, 1); // Pins 6,7,8,9 from Port C are out (0b1111000000)
     BSP_SPI_Init();
+
+    // Initialize txBuf and rxBuf
+    uint8_t txReadBuf[2] = {SPI_OPCODE_R, SPI_IODIRB}; //0x01 is IODIRB in Bank 0 Mode
+    uint8_t rxBuf = 0;
+    
+    // Reads direction register and stores in rxBuf
+    GPIO_WriteBit(GPIOA, GPIO_Pin_4, Bit_RESET);
+    BSP_SPI_Write(txReadBuf, 2);
+    BSP_SPI_Read(rxBuf,1);
```

Comment:
```suggestion
    BSP_SPI_Read(&rxBuf,1);
```
If you change rxBuf to a `uint8_t`, you will have to change this as well